engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeExecutionEnvironmentTest.scala

```diff
@@ -129,6 +129,11 @@ class RuntimeExecutionEnvironmentTest
     val Some(Api.Response(_, Api.InitializedNotification())) = context.receive
   }
 
+  override protected def afterEach(): Unit = {
+    context.executionContext.context.close()
```

Comment:
When I added all these `afterEach` cleanups, running the whole `sbt runtime-with-instruments/test` started to randomly fail. Turned out I had to also f71557db - e.g. wait for the `shutdown` of the `EnsoContext` threads.